### PR TITLE
Nnikolov/gestures refactoring

### PR DESCRIPTION
**Pull request by nsndeck**

### CODE REVIEW

**Comment by enchev** [comment]
:+1:

File: ui/core/view-common.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ export class View extends proxy.ProxyObject implements definition.View {
 
     public _cssClasses: Array<string> = [];
 
-    private _gesturesObserver: gestures.GesturesObserver;
+    public _gestureObservers = {};
```
Comment (ErjanGavalji) [inline ui/core/view-common.ts]: Shall we add a type annotation?

File: ui/core/view-common.ts [hunk 2]
```diff
@@ -147,9 +147,22 @@ export class View extends proxy.ProxyObject implements definition.View {
         this._visualState = visualStateConstants.Normal;
     }
 
-    observe(type: number, callback: (args: gestures.GestureEventData) => void, thisArg?: any): gestures.GesturesObserver {
-        this._gesturesObserver = gestures.observe(this, type, callback, thisArg);
-        return this._gesturesObserver;
+    observe(type: number, callback: (args: gestures.GestureEventData) => void, thisArg?: any): void {
```
Comment (hshristov) [inline ui/core/view-common.ts]: This should be marked as breaking change in the issue and in docs.

File: ui/core/view.d.ts [hunk 3]
```diff
@@ -355,7 +355,7 @@ declare module "ui/core/view" {
          */
         public focus(): boolean;
 
-        observe(type: number, callback: (args: gestures.GestureEventData) => void, thisArg?: any): gestures.GesturesObserver;
```
Comment (hshristov) [inline ui/core/view.d.ts]: breaking change. mark it

File: ui/gestures/gestures.android.ts [hunk 4]
```diff
@@ -70,9 +72,6 @@ export class GesturesObserver implements definition.GesturesObserver {
 
     private _dettach() {
         trace.write(this._target + "._detach() android:" + this._target.android, "gestures");
-        if (this._target && this._target.android) {
```
Comment (hshristov) [inline ui/gestures/gestures.android.ts]: Why not detaching?

**Pull request merged**